provider/terraform/resources/resource_storage_default_object_acl.go

```diff
@@ -2,185 +2,109 @@ package google
 
 import (
 	"fmt"
-	"log"
-
 	"github.com/hashicorp/terraform/helper/schema"
 	"google.golang.org/api/storage/v1"
 )
 
 func resourceStorageDefaultObjectAcl() *schema.Resource {
 	return &schema.Resource{
-		Create:        resourceStorageDefaultObjectAclCreate,
-		Read:          resourceStorageDefaultObjectAclRead,
-		Update:        resourceStorageDefaultObjectAclUpdate,
-		Delete:        resourceStorageDefaultObjectAclDelete,
-		CustomizeDiff: resourceStorageRoleEntityCustomizeDiff,
+		Create: resourceStorageDefaultObjectAclCreateUpdate,
+		Read:   resourceStorageDefaultObjectAclRead,
+		Update: resourceStorageDefaultObjectAclCreateUpdate,
+		Delete: resourceStorageDefaultObjectAclDelete,
 
 		Schema: map[string]*schema.Schema{
-			"bucket": &schema.Schema{
+			"bucket": {
 				Type:     schema.TypeString,
 				Required: true,
 				ForceNew: true,
 			},
 
-			"role_entity": &schema.Schema{
-				Type:     schema.TypeList,
+			"role_entity": {
```

Comment:
Given that passing an empty slice will zero our the roles:
Should `role_entity` be required?
If not, can you update that docs that omitting it will remove acls?

Reply:
Good catch! I didn't remove `Required` from the docs. Updated.